configman/config_manager.py

```diff
@@ -675,7 +686,9 @@ def _overlay_expand(self):
                 if key in all_reference_values:
                     # make sure that this value gets propagated to keys
                     # even if the keys have already been overlaid
-                    known_keys -= set(all_reference_values[key])
+                    finished_keys -= set(
+                        all_reference_values[key]
+                    )
```

Comment:
my objection still stands.  Removing all the ```values``` from finished_keys set will cause an infinite loop.  We want to only consider the reference values from the current key and remove those.